sei-tendermint/types/block.go

```diff
@@ -640,10 +640,11 @@ const (
 
 // CommitSig is a part of the Vote included in a Commit.
 type CommitSig struct {
-	BlockIDFlag      BlockIDFlag `json:"block_id_flag"`
-	ValidatorAddress Address     `json:"validator_address"`
-	Timestamp        time.Time   `json:"timestamp"`
-	Signature        []byte      `json:"signature"`
+	BlockIDFlag BlockIDFlag `json:"block_id_flag"`
+	// WARNING: all fields below should be zeroed if BlockIDFlag == BlockIDFlagAbsent
+	ValidatorAddress Address    `json:"validator_address"`
+	Timestamp        time.Time  `json:"timestamp"`
+	Signature        crypto.Sig `json:"signature"`
```

Comment:
crypto.Sig is of a fixed size. The json marshalling will be backward compatible, as long as it is a non-empty slice.